### ISSUE TITLE
[bug] AutotoolsToolchain regression in 1.45

### ISSUE THREAD

**Issue by SSE4** [conan-io/conan#10640]
from https://github.com/conan-io/conan-center-index/pull/9488
running builds in 1.44.1 vs 1.45.0 leads to different results, 1.44.1 builds successfully, and 1.45.0 fails.

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 16.04.7 LTS (docker images `conanio/gcc5:1.44.1` and `conanio/gcc5:1.45.0`)
  * Compiler+version: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
  * Conan version: Conan version 1.45.0
  * Python version: Python 3.7.5

### Steps to reproduce (Include if Applicable)

run docker image (1.44.1 vs 1.45.0), build libkmod recipe like:
```
git clone --branch kmod_bug --depth 1 git@github.com:SSE4/conan-center-index.git
cd conan-center-index/recipes/kmod/all
conan create . kmod/29@ -k
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

some logs with verbose make enabled (`tc.make_args.append("V=1")`)
[1.44.log](https://github.com/conan-io/conan/files/8123042/1.44.log)
[1.45.log](https://github.com/conan-io/conan/files/8123043/1.45.log)

the main (vital) difference is:
```
-e 's,@libzstd_CFLAGS\@,-I/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c50da0e296272b3/include,g' -e 's,@libzstd_LIBS\@,-L/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc
95249ac3e0c50da0e296272b3/lib -lzstd -lpthread -F /home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c50da0e296272b3/Frameworks,g' 
```
vs
```
-e 's,@libzstd_CFLAGS\@,,g' -e 's,@libzstd_LIBS\@,,g'
```
(later is wrong)

**Comment by memsharded** [comment]
Quick thing: The ``AutotoolsToolchain`` shouldn't be adding ``-I/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c50da0e296272b3/include``, if that was happening in 1.44. This include path should come from ``AutotoolsDeps`` or ``PkgConfigDeps``.

**Comment by SSE4** [comment]
> Quick thing: The `AutotoolsToolchain` shouldn't be adding `-I/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c50da0e296272b3/include`, if that was happening in 1.44. This include path should come from `AutotoolsDeps` or `PkgConfigDeps`.

yes, it's coming from `PkgConfigDeps` indeed.
the thing I have faced trying it, is that `PkgConfigDeps` was broken in 1.44, e.g. generating files referencing themselves:
```
$ cat /home/conan/.conan/data/kmod/29/_/_/build/e7842461bbcd29441fdb340a5299c019eae366ea/libzstd.pc
Name: libzstd
Description: Conan package: libzstd
Version: 1.5.2
Requires: libzstd
```
(seems to be an issue https://github.com/conan-io/conan/issues/10341)
once it was fixed in 1.45, it caused a bad command line:
```
libtool: link: /usr/bin/gcc -shared  -fPIC -DPIC  libkmod/.libs/libkmod.o libkmod/.libs/libkmod-builtin.o libkmod/.libs/libkmod-list.o libkmod/.libs/libkmod-config.o libkmod/.libs/libkmod-index.o libkmod/.libs/libkmod-module.o libkmod/.libs/libkmod-file.o libkmod/.libs/libkmod-elf.o libkmod/.libs/libkmod-signature.o  -Wl,--whole-archive shared/.libs/libshared.a -Wl,--no-whole-archive  -L/home/conan/.conan/data/zstd/1.5.2/_/_/package/ff7e1eda45846fc95249ac3e0c50da0e296272b3/lib -L/home/conan/.conan/data/xz_utils/5.2.5/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -L/home/conan/.conan/data/zlib/1.2.11/_/_/package/76f87539fc90ff313e0b3182641a9bb558a717d2/lib -L/home/conan/.conan/data/openssl/3.0.1/_/_/package/d5204605bf864d4bd98fc6a9e624938e8e04ad03/lib -ldl -lrt -lpthread -lzstd -llzma -lz -lssl -lcrypto  -m64 -O3 -Wl,--version-script=/home/conan/.conan/data/kmod/29/_/_/build/e7842461bbcd29441fdb340a5299c019eae366ea/source_subfolder/libkmod/libkmod.sym -m64 -F -F -F -F -F -F -F -F -F   -Wl,-soname -Wl,libkmod.so.2 -o libkmod/.libs/libkmod.so.2.3.7
```
NOTICE : `-F -F -F -F -F -F -F -F -F   -Wl,-soname -Wl,libkmod.so.2`
it seems like that ate soname argument as a result

(/cc @franramirez688 as he was working on soname, as @czoido as he was working on PkgConfigDeps bug mentioned)
